### PR TITLE
Feature: write local-XC and EXX part of band(orbital) energy along with the XC matrix

### PR DESCRIPTION
**Pull request by maki49**
The orbital energy will be printed in the `OUT.${suffix}/orbital_energy` file along with the XC matrix `OUT.${suffix}/k-${ik}-Vxc` when `out_mat_xc==1`.  
@ minyez @ AroundPeking FYI

### CODE REVIEW

File: source/module_io/write_Vxc.hpp [hunk 1]
```diff
@@ -149,6 +149,25 @@ namespace ModuleIO
...
     }
 
+    template<typename T>
+    std::vector<double> orbital_energy(
+        const int ik,
+        const int nbands,
+        const std::vector<T>& mat_mo,
+        const Parallel_2D& p2d)
+    {
+        std::vector<double> e(nbands, 0.0);
```
Comment (mohanchen) [inline source/module_io/write_Vxc.hpp:162]: I will suggest using "defensive coding", check nbands>=0 when it comes into this function.

File: source/module_io/write_Vxc.hpp [hunk 2]
```diff
@@ -149,6 +149,25 @@ namespace ModuleIO
...
+        for (int i = 0;i < nbands;++i)
+        {
+            if (p2d.in_this_processor(i, i))
+            {
+                e[i] = get_real(mat_mo[p2d.global2local_col(i) * p2d.get_row_size() + p2d.global2local_row(i)]);
```
Comment (mohanchen) [inline source/module_io/write_Vxc.hpp]: I would suggest using a variable to record the index = p2d.global2local_col(i) * p2d.get_row_size() + p2d.global2local_row(i)], and then use mat_mo[index]

Comment (maki49) [reply source/module_io/write_Vxc.hpp]: I've changed it according to your suggestion. But Why? Here "*" and "+" calculations are not repeated so `index` is not reused.

**Review by PeizeLin — APPROVED** [review]

**Review by mohanchen — COMMENTED** [review]
wait till I find a unified output style.

**Review by mohanchen — APPROVED** [review]
LGTM

**Pull request merged**